### ISSUE TITLE
Search method always returns page 1 no matter what page is supplied.

### ISSUE THREAD

**Issue by Macaroni-Biff** [AlexandreSenpai/Enma#61]
As the title suggests, I'm having an issue where the search method is always returning the default page of search results, 1. I'm attempting to use the search method to jump to the 3rd page of the search results. Which is what I assume the page argument is for...

Enma version: 2.2.1
Python version: 3.12
OS: Windows 10 (Though I'm using bash for windows)
Source: This is happening while using the nHentai source. I haven't tested with any other sources.

My code:

```
from typing import cast
from enma import Enma, DefaultAvailableSources, CloudFlareConfig, NHentai, Sort

enma = Enma[DefaultAvailableSources]()
config = CloudFlareConfig(
    user_agent='<my-user-agent>',
    cf_clearance='<my-cf-clearance>'
)

enma.source_manager.set_source(source_name='nhentai')
nh_source = cast(NHentai, enma.source_manager.source)
nh_source.set_config(config=config)

foo = enma.search('tanline', 3)
print(foo)
print('Output: ' + str(foo.page))
```

The printed object (sorry, I didn't take the time to make it prettier. It's late.):

![image](https://github.com/AlexandreSenpai/Enma/assets/156641554/0c44d74b-bf26-49b5-b505-1fabbe7100f5)

Second print result:
`Output: 1`

This will always be the case. No matter what number I supply to the page argument, it is always returned as `1`.

That's the only issue, everything else works perfectly fine. I'm able to search, view and download everything. I just need to get this search pagination implemented and I'll be golden ponyboy!

Please let me know if I'm misunderstanding what the `page` parameter is supposed to be used for, as this could very well be a layer 8 issue. However, if that's the case and this always returns 1 then I don't really see why it's a required value.

Thank you so much for this library and all the hard work put into it!

**Comment by AlexandreSenpai** [comment]
Hello, thanks for reporting. This is a bug indeed. `page` property representa the source pagination value. That means that the `page` must match the source corrent pagination number.

**Comment by AlexandreSenpai** [comment]
Available at version 2.3.0!

```sh
pip install enma==2.3.0
```
